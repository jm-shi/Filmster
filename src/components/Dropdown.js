import React from 'react';
import { NavLink } from 'react-router-dom';

class Dropdown extends React.Component {
  constructor() {
    super();

    this.state = {
      displayMenu: false
    };

    this.toggleMenu = this.toggleMenu.bind(this);
  }

  toggleMenu = e => {
    e.preventDefault();
    this.setState({ displayMenu: !this.state.displayMenu });
  };

  render() {
    return (
      <div className="dropdown">
        <div className="dropdown__button" onClick={this.toggleMenu}>
          Settings
        </div>

        {this.state.displayMenu ? (
          <ul>
            <li>
              <NavLink exact={true} className="link" to="/">
                Home
              </NavLink>
            </li>
            <li>
              <NavLink className="link" to="/search">
                Search
              </NavLink>
            </li>
            <li>
              <NavLink className="link" to="/discover">
                Discover
              </NavLink>
            </li>
          </ul>
        ) : null}
      </div>
    );
  }
}

export default Dropdown;
