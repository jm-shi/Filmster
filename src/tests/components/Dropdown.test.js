import React from 'react';
import { shallow } from 'enzyme';
import Dropdown from '../../components/Dropdown';

test('should render Dropdown', () => {
  const wrapper = shallow(<Dropdown />);
  expect(wrapper).toMatchSnapshot();
});

test('should toggle menu upon click', () => {
  const event = {
    preventDefault() {},
    target: { displayMenu: true }
  };
  const wrapper = shallow(<Dropdown />);
  let displayMenuState = wrapper.state('displayMenu');
  expect(displayMenuState).toEqual(false);
  wrapper.find('.dropdown__button').prop('onClick')(event);
  displayMenuState = wrapper.state('displayMenu');
  expect(displayMenuState).toEqual(true);
});
