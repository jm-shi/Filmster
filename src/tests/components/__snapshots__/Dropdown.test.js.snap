// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Dropdown 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dropdown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="dropdown__button"
          onClick={[Function]}
>
          Settings
</div>,
        null,
      ],
      "className": "dropdown",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Settings",
          "className": "dropdown__button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Settings",
        "type": "div",
      },
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="dropdown__button"
            onClick={[Function]}
>
            Settings
</div>,
          null,
        ],
        "className": "dropdown",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Settings",
            "className": "dropdown__button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Settings",
          "type": "div",
        },
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
